src/Components/Components/src/InjectAttribute.cs

```diff
@@ -10,4 +10,8 @@ namespace Microsoft.AspNetCore.Components;
 [AttributeUsage(AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
 public sealed class InjectAttribute : Attribute
 {
+    /// <summary>
+    /// Gets or sets the object that specifies the key of the service to inject.
+    /// </summary>
+    public object? Key { get; init; }
```

Comment:
Any reason `[FromKeyServices(...)]` can't be used instead of `[Inject]`?

Reply:
It should be a totally possible alternative - it just isn't consistent with the existing Blazor experience. I'm definitely open to considering it as an alternative, though.

Reply:
@BrennanConroy `Inject` is the Blazor/Razor specific attribute that the Razor compiler targets for the `@inject` directive.